### PR TITLE
Upgraded to Font Awesome 5.8.2 (including v4 shims)

### LINKED ISSUE #53: Update to latest FontAwesome
FontAwesome 5.x release is out. They have changed their naming convention so we'll need to ensure that components that reference FA icons get updated as well.

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**
Fixes #53 

Based on https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4, we can include the shim and introduce this without a breaking change.

To test not breaking backward compatibility, you can run `skyux serve` and click on the Fonts link.  I'm showing using old V4 font class names, using V4 custom CSS class, and all 3 free V5 fonts.

### CODE REVIEW

File: src/app/visual/fonts/fonts.component.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+</p>
+
+<ul>
+  <li><span class="icon login"></span> Login</li>
+  <li><span class="icon tps"></span> TPS Reports</li>
+  <li><span class="icon twitter"></span> Twitter</li>
+  <li><i class="fa fa-bank"></i> V4 Bank</li>
+  <li><i class="fas fa-university"></i> V5 FAS - University</li>
+  <li><i class="far fa-chart-bar"></i> V5 FAR - Chart Bar</li>
+  <li><i class="fab fa-amazon-pay"></i> V5 FAB - Amazon Pay</li>
```
Comment (Blackbaud-AlexKingman) [inline src/app/visual/fonts/fonts.component.html]: Hate to be that guy... but can we put each attribute on a new line?
https://host.nxt.blackbaud.com/skyux-team-docs/style#component-html-tags

File: src/app/visual/fonts/fonts.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<p>
+  Fonts: {{ status }}
+</p>
+
+<ul>
```
Comment (Blackbaud-AlexKingman) [inline src/app/visual/fonts/fonts.component.html]: Thanks for creating these. In addition, would we want to create a visual tests for this, or would all problems with fonts be captured in unit?

File: src/app/app-extras.module.ts [hunk 3]
```diff
@@ -10,10 +10,17 @@ import {
...
+  SkyAppStyleLoader
+} from './public/style-loader';
+
 @NgModule({
   exports: [
     SkyThemeModule,
     SkyMediaQueryModule
+  ],
+  providers: [
+    SkyAppStyleLoader
```
Comment (Blackbaud-SteveBrush) [inline src/app/app-extras.module.ts]: The `SkyAppStyleLoader` is provided by the `SkyThemeModule`, so this doesn't need to be added here?

File: src/app/public/style-loader.ts [hunk 4]
```diff
@@ -14,15 +14,16 @@ export class SkyAppStyleLoader {
...
+    const fontAwesome5 = new FontFaceObserver('Font Awesome 5 Brands', { weight: 400 });
     const blackbaudSans = new FontFaceObserver('Blackbaud Sans');
 
     return Promise
       .all([
-        // Specify a character for FontAwesome since some browsers will fail to detect
-        // when the font is loaded unless a known character with a different width
-        // than the default is not specified.
-        fontAwesome.load('\uf0fc', SkyAppStyleLoader.LOAD_TIMEOUT),
+        // FA does not contain the latin characters "BESbwy", so we need a custom test string.
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/style-loader.ts]: What is "BESbwy"?

File: src/app/visual/fonts/fonts.component.scss [hunk 5]
```diff
@@ -0,0 +1,19 @@
+.icon::before {
+  display: inline-block;
+  font-style: normal;
+  font-variant: normal;
+  text-rendering: auto;
+  -webkit-font-smoothing: antialiased;
+}
+
+.login::before {
+  font-family: "Font Awesome 5 Free"; font-weight: 900; content: "\f007";
```
Comment (Blackbaud-SteveBrush) [inline src/app/visual/fonts/fonts.component.scss]: Is the point of this line to "prove" that FontAwesome 5 is registered on the page in all of its forms?

Just curious what this serves in relation to a consumer (they wouldn't be writing "Font Awesome 5 Free", correct)?

File: src/app/public/style-loader.ts (hunk 4 above)
Comment (Blackbaud-BobbyEarl) [reply src/app/public/style-loader.ts]: Explained in the url immediately following the comment.

File: src/app/app-extras.module.ts (hunk 3 above)
Comment (Blackbaud-BobbyEarl) [reply src/app/app-extras.module.ts]: Good catch.  Removed.

File: src/app/visual/fonts/fonts.component.html (hunk 1 above)
Comment (Blackbaud-BobbyEarl) [reply src/app/visual/fonts/fonts.component.html]: I don't mind changing at all, but can you elaborate?  I only see one attribute, do you mean element? Do you want...

```
  <li>
    <span class="icon login"></span> Custom V4 Login
  </li>
  <li>
    <span class="icon tps"></span> Custom V5 (new icon) TPS Reports
  </li>
...
```

File: src/app/visual/fonts/fonts.component.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+.icon::before {
```
Comment (Blackbaud-BobbyEarl) [inline src/app/visual/fonts/fonts.component.scss]: The point of these custom classes are to show that the pseudo-class technique still works for V4 and V5 icons.  It is mostly taken directly from the migration guide as something you'll have to fix from V4 -> V5.  https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4

File: src/app/visual/fonts/fonts.component.html (hunk 2 above)
Comment (Blackbaud-BobbyEarl) [reply src/app/visual/fonts/fonts.component.html]: Good suggestion, I think a visual test would be relevant. I'll add one.

File: src/app/visual/fonts/fonts.component.scss (hunk 5 above)
Comment (Blackbaud-BobbyEarl) [reply src/app/visual/fonts/fonts.component.scss]: The point of these custom classes are to show that the pseudo-class technique still works for V4 and V5 icons. It is mostly taken directly from the migration guide as something you'll have to fix from V4 -> V5. https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4

**Comment by Blackbaud-BobbyEarl** [comment]
After much discussion, I'm going to close this PR.  I hope to document that decision there, so that in the future when we solve it, my newly-gained knowledge will be useful.

**Pull request closed without merging**